src/agent/configuration_parser/src/configuration_parser.cpp

```diff
@@ -121,4 +106,194 @@ namespace configuration
 
         return static_cast<std::time_t>(std::stoul(number) * multiplier);
     }
+
+    bool ConfigurationParser::isValidYamlFile(const std::filesystem::path& configFile) const
+    {
+        try
+        {
+            YAML::Node mapToValidte = YAML::LoadFile(configFile.string());
+            if (!mapToValidte.IsMap() && !mapToValidte.IsSequence())
+            {
+                throw std::runtime_error("The file does not contain a valid YAML structure.");
+            }
+            return true;
+        }
+        catch (const std::exception&)
+        {
+            return false;
+        }
+    }
+
+    void ConfigurationParser::MergeYamlNodes(YAML::Node& base, const YAML::Node& override)
```

Comment:
Maybe this should be separate from the `ConfigurationParser` since it is so yaml specific. Perhaps as a free function utility that we can test.

Reply:
Done!